zomes/service/code/src/servicelog.rs

```diff
@@ -76,14 +75,15 @@ fn _get_original_request(address: Address) -> ZomeApiResult<request::ClientReque
 }
 
 pub fn list_uninvoiced_servicelogs() -> Vec<Address> {
-    // TODO: filter out invoiced ones
-    match hdk::query("service_log".into(), 0, 0) {
+    match handle_list_uninvoiced_servicelogs() {
         Ok(results) => results,
         _ => vec![],
     }
 }
 
-pub fn handle_list_uninvoiced_servicelogs() -> Vec<Address> {
-    list_uninvoiced_servicelogs()
+pub fn handle_list_uninvoiced_servicelogs() -> ZomeApiResult<Vec<Address>> {
+    // List all InvoicedLogs, then join the list of all servicelog_list inside them
+    // TODO: filter out invoiced ones
```

Comment:
We need to filter out any servicelogs that have already been included in an invoice.

Reply:
Hey there, @pjkundert, check this latest commit. It should be working now.

Reply:
Maybe It would be great to include a test for that too, no? :smile: